docs/UserGuide.md

```diff
@@ -161,6 +161,27 @@ Examples:
 * `find olive 87438` returns `87438807`, `Charlotte Oliveiro`<br>
   ![result for 'find olive 87438'](images/findOlive.png)
 
+### Filtering persons by age group or appointment dates: `filter`
+
+Filter persons whose age and/or appointment dates are within the specified range.
+
+Format: `filter [ap/APPOINTMENT_DATE_LOWER_BOUND - APPOINTMENT_DATE_UPPER_BOUND] [b/AGE_LOWER_BOUND - AGE_UPPER_BOUND]`
+
+* The order of the keywords does not matter. 
+* Only appointment dates and/or age group can be used to filter.
+* Dates must be in `dd-MM-yyyy` format.
+* Each value for each field has to be a range (i.e. lower bound - upper bound)
+* At least one field (age or appointment dates) has to be specified.
+* Range are inclusive (i.e. age 79 is considered True in specified range'79-99')
+* Persons must within all specified ranges to be returned if both age and appointment dates are specified (i.e. `AND` search).
+  e.g. `filter b/70-99 ap/01/01/2025 - 01/01/2026` will return `Roy b/87 ap/11/11/2025`.
+
+Examples:
+* `filter b/70-79`
+* `filter ap/01/01/2025 - 01/01/2026`
+* `filter b/70-79 ap/01/01/2025 - 01/01/2026`
+
```

Comment:
There is a command summary table in HelpWindow under ui folder, perhaps you can update it in alphabetical order.

Reply:
Thanks for the reminder!